src/CompositeRheologies.jl

```diff
@@ -529,13 +558,104 @@ This performs nonlinear Newton iterations for `τII` with given `εII_total` for
         error("iterations did not converge")
     end
 
-    τII = x[1]
+    return (x...,)
+end
 
-    return τII
+
+"""
+    local_iterations_εII(c::CompositeRheology{T,N}, εII_total, args)
+
+This performs nonlinear Newton iterations for `τII` with given `εII_total` for cases where we plastic elements
+"""
+@inline function local_iterations_εII(
+    c::CompositeRheology{T,N,
+                        Npar,is_par,            # no ||
+                        Nplast, is_plastic,     # with plasticity
+                        0,is_vol},              # no volumetric
+    εII_total::_T, 
+    args; 
+    tol = 1e-6, 
+    verbose = false,
+    τ_initial = nothing, 
+    ε_init = nothing,
+    max_iter = 1000
+) where {T,N,Npar,is_par, _T, Nplast, is_plastic, is_vol}
+ 
+    # Compute residual
+    n = 1 + Nplast + Npar;             # total size of unknowns
+    x = zero(εII_total)
+
+    # Initial guess of stress & strainrate
+    if isnothing(τ_initial)
+        τ_initial = compute_τII_harmonic(c, εII_total, args)
+    end
+    
+    verbose && println("τII guess = $τ_initial")
+    
+    x    = @MVector zeros(_T, n)
+    x[1] = τ_initial
+
+    j = 1;
+    for i=1:N
+        #if is_par[i]
+        #    j += 1
+        #    x[j] = compute_εII_harmonic_i(c, τ_initial, args,i)   
+        #end
+
+        if is_plastic[i] & is_par[i]
```

Comment:
Should be the short-circuit operator `&&`, same below

Reply:
Done